array-methods.js

```diff
@@ -58,20 +100,119 @@ var stateSums = null;
     round this number to the nearest dollar before moving on.
   )
  */
-var sumOfHighInterests = null;
+var sumOfHighInterests = 0;
+
+let stateSum = [];
+
+dataArray.filter(
+  function(element){
+
+    if (element.state === "WI" || element.state === "IL" || element.state === "WY" ||
+    element.state === "OH" || element.state === "GA" || element.state === "DE"){
+      return false;
+    }
+    return true;
+
+  }
+).forEach(
+  function (dataElement){
+
+    let index = stateSum.findIndex(
+      function(stateSumElement){
+        return stateSumElement.state === dataElement.state;
+      }
+    );
+
+    if (index === -1){
+      stateSum.push({
+        amount : parseFloat(dataElement.amount),
+        state : dataElement.state,
+      })
+      return;
+    }
+
+    stateSum[index].amount += parseFloat(dataElement.amount);
+  }
+)
+
+stateSum.forEach(
+  function (element){
+    element.amount = Math.round(element.amount * .189);
+  }
+);
+
+stateSum.forEach(
+  function (element){
+    if (element.amount > 50000){
+      sumOfHighInterests += element.amount;
+    }
+  }
+);
 
 /*
   set `lowerSumStates` to be an array of two letter state
   abbreviations of each state where the sum of amounts
   in the state is less than 1,000,000
  */
-var lowerSumStates = null;
+
+function sumStates(incomingArr){
```

Comment:
Please update the rest of the file to not create news arrays, or work by calling functions. The functional array methods are all your need to get this assignment done. Also, a hint, forEach is not used needed at all in this file. Lean more toward `.reduce`